e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -896,44 +896,34 @@ public static ProvisioningClientOptions CreateProvisioningClientOptionsFromName(
         private static async Task ConfirmRegisteredDeviceWorksAsync(
             DeviceRegistrationResult result,
             IAuthenticationMethod auth,
-            IotHubClientTransportSettings transportSettings,
-            bool sendReportedPropertiesUpdate)
+            IotHubClientTransportSettings transportSettings)
         {
             await using var iotClient = new IotHubDeviceClient(result.AssignedHub, auth, new IotHubClientOptions(transportSettings));
             VerboseTestLogger.WriteLine("DeviceClient OpenAsync.");
-            await iotClient.OpenAsync().ConfigureAwait(false);
-            VerboseTestLogger.WriteLine("DeviceClient SendTelemetryAsync.");
-
-            var testMessage = new TelemetryMessage("TestMessage");
-            await iotClient.SendTelemetryAsync(testMessage).ConfigureAwait(false);
-
-            if (sendReportedPropertiesUpdate)
+            int attempt = 1;
+            while (true)
             {
-                VerboseTestLogger.WriteLine("DeviceClient updating desired properties.");
-                TwinProperties twin = await iotClient.GetTwinPropertiesAsync().ConfigureAwait(false);
-                var propertiesToReport = new ReportedProperties
+                try
+                {
+                    await iotClient.OpenAsync().ConfigureAwait(false);
+                    break;
+                }
+                catch (Exception ex) when (attempt++ < 4)
```

Comment:
Do we want to retry all scenarios or just the specific auth error that we've been seeing? I think it'd be good to add the comment you have in the PR below into the code comments here.

Reply:
I couldn't identify the auth error from the callstack.

Reply:
We don't print the error code on ToString :(

Reply:
Can we guess what it was? Maybe `IotHubClientErrorCode.Unauthorized`?

Reply:
Yes, it should be IotHubClientErrorCode.Unauthorized.

https://github.com/Azure/azure-iot-sdk-csharp/blob/317b9a3e4558f871ee31b9088c8b9222bde360e0/iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs#L317-L323